tests/subsys/net/lib/nrf_cloud/cloud/src/main.c

```diff
@@ -0,0 +1,1880 @@
+ * in dc_connected state with the topic_type is equal
+ * topic_bulk and nct_dc_bulk_send succeeds
+ */
+ZTEST(nrf_cloud_test, test_cloud_send_topic_bulk_success)
+{
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Init the cloud successfully */
+	init_cloud_success();
+
+	/* Connect to the cloud successfully with dc_connect */
+	connect_cloud_dc_success();
+
+	nct_dc_bulk_send_fake.custom_fake = fake_nct_dc_bulk_send__succeeds;
+
+	/* Message data */
+	struct nrf_cloud_tx_data tx_data_t = {
+		.topic_type = NRF_CLOUD_TOPIC_BULK,
+		.qos = MQTT_QOS_0_AT_MOST_ONCE
+	};
+
+	int ret = nrf_cloud_send(&tx_data_t);
+
+	zassert_ok(ret, "return should be 0 when success");
+}
+
+/* Verify that the function will fail if an invalid topic_type is given */
+ZTEST(nrf_cloud_test, test_cloud_send_invalid_topic_type)
+{
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Message data */
+	struct nrf_cloud_tx_data tx_data_t = {
+		.topic_type = 0,
+		.qos = MQTT_QOS_0_AT_MOST_ONCE
+	};
+
+	int ret = nrf_cloud_send(&tx_data_t);
+
+	zassert_equal(-ENODATA, ret, "return should be -ENODATA with invalid topic type");
+}
+
+#if defined(CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD)
+
+/* TEST START_CONNECTION_POLL */
+
+/* Verify that the function will fail
+ * if the cloud is in the idle state
+ */
+ZTEST(nrf_cloud_test, test_start_connection_poll_in_idle)
+{
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	int ret = start_connection_poll();
+
+	zassert_equal(-EACCES, ret, "return should be -EACCES when in idle state");
+}
+
+/* Verify that the function will fail
+ * if the poll thread is already active
+ */
+ZTEST(nrf_cloud_test, test_start_connection_poll_already_active)
+{
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Init the cloud successfully */
+	init_cloud_success();
+
+	zassert_equal(STATE_INITIALIZED, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the initialized state after initialization");
+	(void)start_connection_poll();
+
+	int ret = start_connection_poll();
+
+	zassert_equal(-EINPROGRESS, ret, "return should be -EINPROGRESS when already active");
+}
+
+/* Verify that the function will succeed
+ * if the cloud is successfully initialized
+ */
+ZTEST(nrf_cloud_test, test_start_connection_poll_success)
+{
+	zassert_equal(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	/* Init the cloud successfully */
+	init_cloud_success();
+
+	int ret = start_connection_poll();
+
+	zassert_ok(ret, "return should be 0 when success");
+}
+
+/* TEST NRF_CLOUD_RUN */
```

Comment:
By the same token, this could go in either the MQTT-mocked or non-MQTT-mocked suite

Reply:
But reading some of the tests, MQTT-mocked just makes more sense to me. Better to simulate actual network failure than simulate how nct would react to network failure. Right now these tests are really dependent on a very specific implementation of the nct/fsm